src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/PlatformSwitcherEditor.cs

```diff
@@ -40,20 +45,84 @@ public override void OnInspectorGUI()
+
+                PlayerSettings.Android.minSdkVersion = AndroidSdkVersions.AndroidApiLevel24;
+                PlayerSettings.Android.forceInternetPermission = true;
+                PlayerSettings.Android.forceSDCardPermission = true;
+                PlayerSettings.Android.ARCoreEnabled = true;
+
+                DeployAndValidateAndroidManifest();
             }
 
             // Editor button for iOS platform and functionality
             if (GUILayout.Button("iOS", GUILayout.Height(_buttonHeight)))
             {
                 EditorUserBuildSettings.SwitchActiveBuildTarget(BuildTargetGroup.iOS, BuildTarget.iOS);
-                PlayerSettings.iOS.cameraUsageDescription = "Camera required for AR Foundation";
+
                 PlayerSettings.defaultInterfaceOrientation = UIOrientation.Portrait; // Currently needed based on Marker Visual logic
+
                 PlayerSettings.SetArchitecture(BuildTargetGroup.iOS, 1); // Set Architecture to ARM64
+                PlayerSettings.iOS.targetOSVersionString = "11.0";
+                // TODO: figure out how to programmatically check the "Requires ARKit Support" box (corresponding to "iOSRequireARKit" in ProjectSettings.asset).
+                PlayerSettings.iOS.cameraUsageDescription = "Camera required for AR Foundation";
             }
 
             GUILayout.EndVertical();
         }
+
+        private void DeployAndValidateAndroidManifest()
+        {
+            var assetsDirectoryPath = Application.dataPath.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+
+            string sourceManifestPath = Path.Combine(assetsDirectoryPath, "MixedReality-SpectatorView", "SpectatorView", "Scripts", "ScreenRecording", "Plugins", "Android", "AndroidManifestTemplate.xml");
+
+            string destManifestDirectoryPath = Path.Combine(assetsDirectoryPath, "Plugins", "Android");
+            string destManifestPath = Path.Combine(destManifestDirectoryPath, "AndroidManifest.xml");
+
+            try
+            {
+                if (File.Exists(destManifestPath))
+                {
+                    Debug.Log($"Android manifest \"{destManifestPath}\" already exists. Not copying from \"{sourceManifestPath}\".", this);
+                }
+                else
+                {
+                    Debug.Log($"Copying android manifest \"{sourceManifestPath}\" to \"{destManifestPath}\".", this);
+
+                    Directory.CreateDirectory(destManifestDirectoryPath);
+                    File.Copy(sourceManifestPath, destManifestPath);
+                }
+
+                var manifest = XElement.Load(destManifestPath);
+                var activities = manifest.XPathSelectElements("application/activity").ToArray();
+
+                if (activities.Length != 1)
+                {
+                    throw new System.Exception($"Expected 1 application activity, but got {activities.Length}.");
```

Comment:
maybe log error compared to throw if this is ran in the editor?

Reply:
I need to try/catch to make sure the potential disk operation exceptions get caught and logged with appropriate context, so I figured I'd throw for all errors and catch all errors (including these) and apply the same context to the logging message in the catch block.